crates/apollo-router-core/src/query_planner/model.rs

```diff
@@ -313,12 +182,4 @@ mod tests {
             vec!["test_variable"]
         );
     }
-
-    #[test]
-    fn query_plan_into_json() {
-        assert_eq!(
-            serde_json::to_value(query_plan()).unwrap(),
-            serde_json::from_str::<Value>(qp_json_string().as_str()).unwrap()
-        );
-    }
```

Comment:
Test removed entirely because it asserts that the query plan can be parsed and re-serialized identically. While this is interesting if we needed to serialize the plan, in this case we don't need to and we are validating that this feature (serialization of queryplan) works despite having no use of it. This is maintenance burden.